ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs

```diff
@@ -198,6 +250,127 @@ public void SetFontStyle(string fontStyleString)
+
+        /// <summary>
+        /// Sets the letter spacing for the node.
+        /// </summary>
+        /// <param name="letterSpacing">The letter spacing.</param>
+        [ReactProp(ViewProps.LetterSpacing, DefaultDouble = FontSizeUnset)]
+        public void SetLetterSpacing(int? letterSpacing)
+        {
+            if (_isVirtual)
+            {
+                ThrowException("letterSpacing");
+            }
+
+            var spacing = letterSpacing == null ? 0 : letterSpacing;
+
+            if (_letterSpacing != spacing)
+            {
+                _letterSpacing = spacing;
+                MarkUpdated();
+            }
+        }
+
+        /// <summary>
+        /// Sets the line height.
+        /// </summary>
+        /// <param name="lineHeight">The line height.</param>
+        [ReactProp(ViewProps.LineHeight)]
+        public virtual void SetLineHeight(double? lineHeight)
+        {
+            if (_isVirtual)
+            {
+                ThrowException("lineHeight");
+            }
+
+            var height = lineHeight == null ? 0 : lineHeight;
+
+            if (_lineHeight != height)
+            {
+                _lineHeight = height;
+                MarkUpdated();
+            }
+        }
+
+        /// <summary>
+        /// Sets the maximum number of lines.
+        /// </summary>
+        /// <param name="numberOfLines">Max number of lines.</param>
+        [ReactProp(ViewProps.NumberOfLines)]
+        public virtual void SetNumberOfLines(int? numberOfLines)
+        {
+            if (_isVirtual)
+            {
+                ThrowException("numberOfLines");
+            }
+
+            var lines = numberOfLines == null ? 0 : numberOfLines;
+
+            if (_numberOfLines != lines)
+            {
+                _numberOfLines = lines;
+                MarkUpdated();
+            }
+        }
+
+        /// <summary>
+        /// Sets the text alignment.
+        /// </summary>
+        /// <param name="textAlign">The text alignment string.</param>
+        [ReactProp(ViewProps.TextAlign)]
+        public void SetTextAlign(string textAlign)
+        {
+            if (_isVirtual)
+            {
+                ThrowException("textAlign");
+            }
+
+            var textAlignment = textAlign == "auto" ? 
+                TextAlignment.DetectFromContent :
+                EnumHelpers.ParseNullable<TextAlignment>(textAlign);
+
+            if (_textAlignment != textAlignment)
+            {
+                _textAlignment = textAlignment;
+                MarkUpdated();
+            }
+        }
+
+        /// <summary>
+        /// Sets the text decoration.
+        /// </summary>
+        /// <param name="textDecoration">The text decoration string.</param>
+        [ReactProp(ViewProps.TextDecorationLine)]
+        public void SetTextDecoration(string textDecoration)
+        {
+            var underline = textDecoration != null && textDecoration.Contains("underline") ? true : false;
```

Comment:
I think this can be simplified to:

var underline = textDecoration != null && textDecoration.Contains("underline");

or...

var underline = textDecoration?.Contains("underline") ?? false;